Signal/src/views/GroupTableViewCell.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  Copyright (c) 2017 Open Whisper Systems. All rights reserved.
+//
+
+import UIKit
+
+@objc class GroupTableViewCell: UITableViewCell {
+
+    let TAG = "[GroupTableViewCell]"
+
+    private let avatarView = AvatarImageView()
+    private let nameLabel = UILabel()
+    private let subtitleLabel = UILabel()
+
+    init() {
+        super.init(style: .default, reuseIdentifier: TAG)
+
+        self.contentView.addSubview(avatarView)
+
+        let textContainer = UIView.container()
+        textContainer.addSubview(nameLabel)
+        textContainer.addSubview(subtitleLabel)
+        self.contentView.addSubview(textContainer)
+
+        // Font config
+        nameLabel.font = UIFont.ows_dynamicTypeBody()
+        subtitleLabel.font = UIFont.ows_footnote()
+        subtitleLabel.textColor = UIColor.ows_darkGray()
+
+        // Listen to notifications...
+        // TODO avatar, group name change, group membership change, group member name change
+
+        // Layout
+
+        nameLabel.autoPinEdgesToSuperviewEdges(with: UIEdgeInsets.zero, excludingEdge: .bottom)
+        subtitleLabel.autoPinEdgesToSuperviewEdges(with: UIEdgeInsets.zero, excludingEdge: .top)
+        subtitleLabel.autoPinEdge(.top, to: .bottom, of: nameLabel)
+
+        avatarView.autoPinLeadingToSuperview()
+        avatarView.autoVCenterInSuperview()
+        avatarView.autoSetDimension(.width, toSize: CGFloat(kContactTableViewCellAvatarSize))
+        avatarView.autoPinToSquareAspectRatio()
+
+        textContainer.autoPinEdge(.leading, to: .trailing, of: avatarView, withOffset: kContactTableViewCellAvatarTextMargin)
+        textContainer.autoPinEdge(toSuperviewEdge: .trailing)
+        textContainer.autoVCenterInSuperview()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    public func configure(thread: TSGroupThread, contactsManager: OWSContactsManager) {
+        self.nameLabel.text = thread.groupModel.groupName
+
+        let groupMemberIds: [String] = thread.groupModel.groupMemberIds
+        let groupMemberNames = groupMemberIds.map { (recipientId: String) in
```

Comment:
Should we sort the group member names?

Reply:
Probably... We should also probably use the same sorting in the Group's conversation settings "Group Members" list. I'e added a card.